target/linux/qualcommax/ipq60xx/base-files/lib/upgrade/platform.sh

```diff
@@ -4,6 +4,79 @@ REQUIRE_IMAGE_METADATA=1
 RAMFS_COPY_BIN='fw_printenv fw_setenv head'
 RAMFS_COPY_DATA='/etc/fw_env.config /var/lock/fw_printenv.lock'
 
+remove_oem_ubi_volume() {
+	local oem_volume_name="$1"
+	local oem_ubivol
+	local mtdnum
+	local ubidev
+
+	mtdnum=$(find_mtd_index "$CI_UBIPART")
+	if [ ! "$mtdnum" ]; then
+		return
+	fi
+
+	ubidev=$(nand_find_ubi "$CI_UBIPART")
+	if [ ! "$ubidev" ]; then
+		ubiattach --mtdn="$mtdnum"
+		ubidev=$(nand_find_ubi "$CI_UBIPART")
+	fi
+
+	if [ "$ubidev" ]; then
+		oem_ubivol=$(nand_find_volume "$ubidev" "$oem_volume_name")
+		[ "$oem_ubivol" ] && ubirmvol "/dev/$ubidev" --name="$oem_volume_name"
+	fi
+}
+
+tplink_get_boot_part() {
+	local cur_boot_part
+	local args
+
+	# Try to find rootfs from kernel arguments
+	read -r args < /proc/cmdline
+	for arg in $args; do
+		local ubi_mtd_arg=${arg#ubi.mtd=}
+		case "$ubi_mtd_arg" in
+		rootfs|rootfs_1)
+			echo "$ubi_mtd_arg"
+			return
+		;;
+		esac
+	done
+
+	# Fallback to u-boot env (e.g. when running initramfs)
+	cur_boot_part="$(/usr/sbin/fw_printenv -n tp_boot_idx)"
+	case $cur_boot_part in
+	1)
+		echo rootfs_1
+		;;
+	0|*)
+		echo rootfs
+		;;
+	esac
+}
+
+tplink_do_upgrade() {
```

Comment:
I think that EAP660 uses the same logic, so that can be moved into one file to get reused

Reply:
Commit 9b336455663 took this code that I wrote without attribution. I am not only quite unhappy about that, I also think it is unreasonable to be asked to do more work here.

Reply:
I am following this pull request because I'm running an EAP610-Outdoor and would love to use OpenWRT on it. So, thanks a lot @mrnuke for working on this :)

Regarding the usage of code without attribution, if it helps, I noticed that at least it is mentioned by the author of that other commit in a post in the respective pull request, https://github.com/openwrt/openwrt/pull/15832#issuecomment-2197803859:

> The new TP-Link factory image format is currently a work in progress. Some scripts were copied from EAP610-Outdoor. See #14922.

Although I definitely agree that it would be better to have an attribution in the commit message for the attribution to be actually visible.